lib/cpp/src/thrift/concurrency/ThreadManager.cpp

```diff
@@ -339,103 +357,88 @@ void ThreadManager::Impl::addWorker(size_t value) {
     idMap_.insert(std::pair<const Thread::id_t, shared_ptr<Thread> >((*ix)->getId(), *ix));
```

Comment:
In addition, if you look at the logic around canSleep and the use of a timeout, it isn't even correct.  It checks to see if a timeout was requested then proceeds into an infinite loop waiting for a slot to add the task.